### PR TITLE
feat: added Switch widget and documentation.

### LINKED ISSUE #3210: Explicit Switch widget and disableOnEdit for boolean values
**Description**

Explicitly add the `Switch`  as a widget in the list of Simple form widgets and add `disableOnEdit` functionality.

**Reasons**

Currently the only way to use the `Switch` / `SwitchRenderer` is to let busola infer based on value (string, number, integer, etc), but I don't see anywhere in the documentation that it does that.

`disableOnEdit` is already implemented for `Text` widgets in the case the user have immutable values (values set only on creation). But boolean values don't have this functionality.

**Attachments**

### PR DESCRIPTION
**Pull request by kevin-kho**
**Description**

Changes proposed in this pull request:

- Explicit Switch widget and schema
- `disableOnEdit` functionality with the Switch widget
- Documentation for the above.

**Related issue(s)**

Fixes #3210 

**Definition of done**

- [x] The PR's title starts with one of the following prefixes:
  - feat: A new feature
  - fix: A bug fix
  - docs: Documentation only changes
  - refactor: A code change that neither fixes a bug nor adds a feature
  - test: Adding tests
  - chore: Maintainance changes to the build process or auxiliary tools, libraries, workflows, etc.
- [x] Related issues are linked. To link internal trackers, use the issue IDs like `backlog#4567`
- [x] Explain clearly why you created the PR and what changes it introduces
- [x] All necessary steps are delivered, for example, tests, documentation, merging

### CODE REVIEW

File: docs/extensibility/60-form-widgets.md [hunk 1]
```diff
@@ -72,6 +73,32 @@ See the following examples:
...
 <img src="./assets/form-widgets/Dropdown2.png" alt="Example of a dropdown text widget" style="border: 1px solid #D2D5D9">
 
+### `Switch`
+
+The `Switch` widgets render a switch button that is used to control boolean values.
+
+| Parameter         | Required | Type    | Description                                                                                                                                                             |
+| ----------------- | -------- | ------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| **description**   | No       | string  | A string displayed in a tooltip when you hover over a question mark icon, next to the input's label. The default value is taken from the CustomResourceDefintion (CRD). |
+| **disableOnEdit** | No       | boolean | Disables a field in edit mode, defaults to `false`.                                                                                                                     |
```
Comment (grego952) [inline docs/extensibility/60-form-widgets.md:83]: Are we talking about any field or a particular one?

Comment (grego952) [inline docs/extensibility/60-form-widgets.md:83]:
```suggestion
| **disableOnEdit** | No       | boolean | Disables a switch in edit mode, defaults to `false`.                                                                                                                     |
```

File: public/schemas/schema-form.yaml [hunk 2]
```diff
@@ -312,3 +312,17 @@ $widgets:
...
+          const: Switch
+    then:
+      properties:
+        description:
+          type: string
+          description: A string displayed in a tooltip when you hover over a question mark icon, next to the input's label. The default value is taken from the CustomResourceDefintion (CRD).
+        disableOnEdit:
+          type: boolean
+          description: a boolean which specifies if field is disabled on edit.
+          default: false
```
[the remark this answers is not in the document]
Comment (kevin-kho) [reply public/schemas/schema-form.yaml]: addressed in [dff6761](https://github.com/kyma-project/busola/pull/3211/commits/dff6761cebde135267d1707f57c696115604f0e4)

File: src/components/Extensibility/components-form/SwitchRenderer.js [hunk 3]
```diff
@@ -36,6 +37,7 @@ export function SwitchRenderer({
       data-testid={storeKeys.join('.') || tFromStoreKeys(storeKeys, schema)}
       input={Inputs.Switch}
       compact={compact}
+      disabled={disableOnEdit}
```
[the remark this answers is not in the document]
Comment (kevin-kho) [reply src/components/Extensibility/components-form/SwitchRenderer.js]: addressed in [a7c3a09](https://github.com/kyma-project/busola/pull/3211/commits/a7c3a09f0e82f38c9512ffe5e4a79538ba5f7428)

File: docs/extensibility/60-form-widgets.md (hunk 1 above)
Comment (kevin-kho) [reply docs/extensibility/60-form-widgets.md]: addressed in [a7c3a09](https://github.com/kyma-project/busola/pull/3211/commits/a7c3a09f0e82f38c9512ffe5e4a79538ba5f7428)

**Review by mrCherry97 — APPROVED** [review]

**Review by grego952 — APPROVED** [review]

**Pull request merged**